sphinx/environment/__init__.py

```diff
@@ -677,21 +677,37 @@ def read_doc(self, docname, app=None):
         language = self.config.language or 'en'
         self.settings['language_code'] = language
         if 'smart_quotes' not in self.settings:
-            self.settings['smart_quotes'] = True
+            self.settings['smart_quotes'] = self.config.smart_quotes
```

Comment:
now that Sphinx core has a ``smart_quotes`` config setting, possibly the logic for

    if 'smart_quotes' not in self.settings:

(see #4112) is gone? can an extension override a config setting rather than add it to environment as done at  https://github.com/sphinx-contrib/spelling/commit/6209fa8d2c16182b30053f40770dc81b11c77f00 ?

Reply:
ok, I will leave it. I find "BuildEnvironment" complicated...

Reply:
Absolutely, `BuildEnvironment` has multiple responsibility. So I'd like to split it into some modules in future.